pkg/machine/e2e/machine_test.go

```diff
@@ -62,7 +63,13 @@ var _ = BeforeSuite(func() {
 	if pullError != nil {
 		Fail(fmt.Sprintf("failed to pull wsl disk: %q", pullError))
 	}
-
+	if testProvider.VMType() == define.AppleHvVirt {
+		cmd := exec.Command("softwareupdate", "--install-rosetta", "--agree-to-license")
```

Comment:
Does this download anything?
I rather have this not be part of the actual test code and part of the CI setup somewhere.
@cevich How easy is this do add to the macos runners by default.

Reply:
Yes, download a small file. This requires an Internet connection.
https://support.apple.com/en-us/102527
```
% du -sh /Library/Updates/Rosetta/SoftwareUpdate/RosettaUpdateAuto.pkg
380K	/Library/Updates/Rosetta/SoftwareUpdate/RosettaUpdateAuto.pkg
```
I agree with including it in the CI setup.

Reply:
As far as I have tried, install-rosetta does not require sudo.
I don't know how often Apple updates, but I am concerned about doing a CI setup for every update of the rosetta pkg.

Reply:
Well I guess the answer is simple, the CI currently passes because that command works just fine as user. I assume you only added this because CI failed without it @tnk4on?

So yes in this case I think we really should add this this to the runner setup, because I Really do not want this in the test suite. This stuff can be run locally by users so we should never install dependencies without their knowledge.
I think we we can add it to contrib/cirrus/mac_runner.sh temporarily until we have it in the proper runner seup linked by @cevich

Reply:
> Well I guess the answer is simple, the CI currently passes because that command works just fine as user. I assume you only added this because CI failed without it @tnk4on?

Yes. I was debugging this PR's CI error and realized the cause and added the code.

Reply:
> I think we we can add it to contrib/cirrus/mac_runner.sh

Keep in mind these Mac's are shared by multiple tasks and multiple PRs (not at the same time).  If you add it into that script, maybe wrap it in a condition that checks if it's already installed first?

> doing a CI setup for every update of the rosetta pkg.

The Macs servicing CI are recreated roughly every 24 hours.  If/when it's installed in the big `setup.sh` script, any updates would be picked up over time.  This isn't ideal, since multiple runs of a PR could net different results depending on which Mac you get.

Reply:
@Luap99 @cevich 
The other code fixes for this PR are complete, waiting for mac CI setup changes.
What will happen regarding CI setup changes?

Reply:
I suggest you move this command into `contrib/cirrus/mac_runner.sh` for now. Possibly checking if it is already installed before calling this command.

I don't like this being in the test code.

Reply:
We can work on a proper integration in the automation repo later but for now we should get this merged

Reply:
I opened a Jira card to track this.